dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/supplier/ProgramStageInstanceProgramStageMapSupplier.java

```diff
@@ -56,12 +56,13 @@
 
     private static final String PI_UID = "programInstanceUid";
 
-    private static final String SQL = "select ps.uid as " + PS_UID + ", pi.uid as " + PI_UID + " " +
+    private static final String SQL = "SELECT DISTINCT ps.uid AS " + PS_UID + ", pi.uid AS " + PI_UID + " " +
         " from programstage AS ps " +
         " JOIN programinstance AS pi ON pi.programid = ps.programid " +
-        " where exists( select programstageinstanceid from programstageinstance psi where  psi.deleted = false " +
-        " and psi.status != 'SKIPPED' and ps.programstageid = psi.programstageid " +
-        " and pi.programinstanceid = psi.programinstanceid ) ";
+        " JOIN programstageinstance AS psi ON pi.programinstanceid = psi.programinstanceid " +
```

Comment:
Looks. I think we generally use lowercase for keywords in SQL statements in Java, would be nice to stick with that.